### ISSUE TITLE
whitespace rule false positive in JSX elements

### ISSUE THREAD

**Issue by adidahiya** [palantir/tslint#559]
reported by @myitcv:

Consider the following:

```typescript
/// <reference path="../../tsd.d.ts" />

import * as React from "react";

interface IProps {
   foo: string;
}

class MyComponent extends React.Component<IProps, {}> {
   render(): JSX.Element {
      return <span>{ this.props.foo }</span>;
   }
}

export function BuildMyComponent(): JSX.Element {
   let x: string = "test";
   return <MyComponent foo={ x } />;
}
```

I get the following whitespace warning:

```
test.tsx[17, 26]: missing whitespace
```

Line 17 is the return statement in the exported function.

**Comment by myitcv** [comment]
Thanks @adidahiya I'll raise as separate issues in future

**Comment by myitcv** [comment]
Thanks very much!
